pandas/io/tests/test_parsers.py

```diff
@@ -2664,6 +2664,40 @@ def test_empty_header_read(count):
         for count in range(1, 101):
             test_empty_header_read(count)
 
+    def test_uneven_lines_with_usecols(self):
+        # See gh-12203
+        csv = r"""a,b,c
+        0,1,2
+        3,4,5,6,7
+        8,9,10
+        """
+
+        # make sure that an error is still thrown
+        # when the 'usecols' parameter is not provided
+        name = self.__class__.__name__
+        msg = "Expected \d+ fields in line \d+, saw \d+"
```

Comment:
this is too much gymnastics here. Why is this needed? e.g. diffs in the errors from python/c parser? if that IS the case. then I'd like to change the raise for this kind of error to a ``ValueError`` in the cparser.